receiver/prometheusreceiver/internal/metadata.go

```diff
@@ -37,18 +37,18 @@ type metadataService struct {
 }
 
 func (s *metadataService) Get(job, instance string) (MetadataCache, error) {
-	targetGroup, ok := s.sm.TargetsAll()[job]
-	if !ok {
-		return nil, errors.New("unable to find a target group with job=" + job)
-	}
-
-	// from the same targetGroup, instance is not going to be duplicated
-	for _, target := range targetGroup {
-		if target.Labels().Get(model.InstanceLabel) == instance {
-			return &mCache{target}, nil
+	for originalJobLabel, targetGroup := range s.sm.TargetsAll() {
+		for _, target := range targetGroup {
+			switch {
+			case target.Labels().Get(model.InstanceLabel) == instance && originalJobLabel == job:
+				// job label was not rewritten
```

Comment:
Since this will be the common case, can we handle it with the previous direct lookup method and fall back to the iterative search only if it isn't found?

Reply:
Totally - I'll take a look at refactoring to apply this optimization

Reply:
@dashpole @rakyll  updated